src/java.base/share/classes/jdk/internal/foreign/StringSupport.java

```diff
@@ -46,154 +40,158 @@ public final class StringSupport {
         return new String(bytes, charset);
     }
 
-    private static void writeFast_byte(MemorySegment segment, long offset, Charset charset, String string) {
+    private static void writeByte(MemorySegment segment, long offset, Charset charset, String string) {
         byte[] bytes = string.getBytes(charset);
         MemorySegment.copy(bytes, 0, segment, JAVA_BYTE, offset, bytes.length);
         segment.set(JAVA_BYTE, offset + bytes.length, (byte)0);
     }
 
-    private static String readFast_short(MemorySegment segment, long offset, Charset charset) {
-        long len = strlen_short(segment, offset);
+    private static String readShort(MemorySegment segment, long offset, Charset charset) {
+        long len = chunkedStrlenShort(segment, offset);
         byte[] bytes = new byte[(int)len];
         MemorySegment.copy(segment, JAVA_BYTE, offset, bytes, 0, (int)len);
         return new String(bytes, charset);
     }
 
-    private static void writeFast_short(MemorySegment segment, long offset, Charset charset, String string) {
+    private static void writeShort(MemorySegment segment, long offset, Charset charset, String string) {
         byte[] bytes = string.getBytes(charset);
         MemorySegment.copy(bytes, 0, segment, JAVA_BYTE, offset, bytes.length);
         segment.set(JAVA_SHORT, offset + bytes.length, (short)0);
     }
 
-    private static String readFast_int(MemorySegment segment, long offset, Charset charset) {
-        long len = strlen_int(segment, offset);
+    private static String readInt(MemorySegment segment, long offset, Charset charset) {
+        long len = strlenInt(segment, offset);
         byte[] bytes = new byte[(int)len];
         MemorySegment.copy(segment, JAVA_BYTE, offset, bytes, 0, (int)len);
         return new String(bytes, charset);
     }
 
-    private static void writeFast_int(MemorySegment segment, long offset, Charset charset, String string) {
+    private static void writeInt(MemorySegment segment, long offset, Charset charset, String string) {
         byte[] bytes = string.getBytes(charset);
         MemorySegment.copy(bytes, 0, segment, JAVA_BYTE, offset, bytes.length);
         segment.set(JAVA_INT, offset + bytes.length, 0);
     }
 
-    private static int native_strlen_byte(MemorySegment segment, long start) {
-        // Heap segments must be handled by Java code
-        if (!segment.isNative()) {
-            return strlen_byte(segment, start);
-        }
-
-        if (start > 0) {
-            segment = segment.asSlice(start);
+    /**
+     * {@return the shortest distance beginning at the provided {@code start}
+     *  to the encountering of a zero byte in the provided {@code segment}}
+     * <p>
+     * The method divides the region of interest into three distinct regions:
+     * <ul>
+     *     <li>head (un-aligned access handling on a byte-by-byte basis) (if any)</li>
+     *     <li>body (long aligned access handling eight bytes at a time) (if any)</li>
```

Comment:
This and other comments in this class refer to alignment properties which I don't think are strictly valid. You might want to search for them and adjust them/remove them.